src/param.rs

```diff
@@ -357,59 +360,20 @@ impl Params {
         self.get(key).and_then(|s| s.parse().ok())
     }
 
-    /// Gets the given parameter and parse as [ParamsFile].
-    ///
-    /// See also [Params::get_blob] and [Params::get_path] which may
-    /// be more convenient.
-    pub fn get_file<'a>(&self, key: Param, context: &'a Context) -> Result<Option<ParamsFile<'a>>> {
-        let val = match self.get(key) {
-            Some(val) => val,
-            None => return Ok(None),
-        };
-        ParamsFile::from_param(context, val).map(Some)
-    }
-
-    /// Gets the parameter and returns a [BlobObject] for it.
-    ///
-    /// This parses the parameter value as a [ParamsFile] and than
-    /// tries to return a [BlobObject] for that file.  If the file is
-    /// not yet a valid blob, one will be created by copying the file.
-    ///
-    /// Note that in the [ParamsFile::FsPath] case the blob can be
-    /// created without copying if the path already refers to a valid
-    /// blob.  If so a [BlobObject] will be returned.
-    pub async fn get_blob<'a>(
-        &self,
-        key: Param,
-        context: &'a Context,
-    ) -> Result<Option<BlobObject<'a>>> {
-        let val = match self.get(key) {
-            Some(val) => val,
-            None => return Ok(None),
-        };
-        let file = ParamsFile::from_param(context, val)?;
-        let blob = match file {
-            ParamsFile::FsPath(path) => BlobObject::new_from_path(context, &path).await?,
-            ParamsFile::Blob(blob) => blob,
+    /// Returns a [BlobObject] for the [Param::File] parameter.
+    pub fn get_file_blob<'a>(&self, context: &'a Context) -> Result<Option<BlobObject<'a>>> {
+        let Some(val) = self.get(Param::File) else {
+            return Ok(None);
         };
+        debug_assert!(val.starts_with("$BLOBDIR/"));
```

Comment:
If the function returns `Result`, why not `ensure!` then? The check isn't expensive anyway